core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,332 @@
+                'Expected score_category to be a string, received %s' % type(
+                    self.score_category))
+
+        if (
+                suggestion_models.SCORE_CATEGORY_DELIMITER not in
+                self.score_category):
+            raise utils.ValidationError(
+                'Expected score_category to be of the form'
+                ' score_type%sscore_sub_type, received %s' % (
+                    suggestion_models.SCORE_CATEGORY_DELIMITER,
+                    self.score_category))
+
+        if (
+                len(self.score_category.split(
+                    suggestion_models.SCORE_CATEGORY_DELIMITER))) != 2:
+            raise utils.ValidationError(
+                'Expected score_category to be of the form'
+                ' score_type%sscore_sub_type, received %s' % (
+                    suggestion_models.SCORE_CATEGORY_DELIMITER,
+                    self.score_category))
+
+        if self.get_score_type() not in suggestion_models.SCORE_TYPE_CHOICES:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be among allowed'
+                ' choices, received %s' % self.get_score_type())
+
+    def accept(self):
+        """Accepts the suggestion. Each subclass must implement this function.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement accept.')
+
+    def update_change_cmd_before_accept(self):
+        """Before accepting the suggestion, some parts of the change_cmd might
+        need to be updated. Each subclass must implement this function as
+        applicable.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement '
+            'update_change_cmd_before_accept.')
+
+
+class SuggestionEditStateContent(BaseSuggestion):
+    """Domain object for a suggestion of type
+    SUGGESTION_TYPE_EDIT_STATE_CONTENT.
+    """
+
+    def __init__( # pylint: disable=super-init-not-called
+            self, suggestion_id, target_id, target_version_at_submission,
+            status, author_id, assigned_reviewer_id, final_reviewer_id,
+            change_cmd, score_category):
+        """Initializes a Suggestion object of type
+        SUGGESTION_TYPE_EDIT_STATE_CONTENT.
+        """
+        self.suggestion_id = suggestion_id
+        self.suggestion_type = (
+            suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT)
+        self.target_type = suggestion_models.TARGET_TYPE_EXPLORATION
+        self.target_id = target_id
+        self.target_version_at_submission = target_version_at_submission
+        self.status = status
+        self.author_id = author_id
+        self.assigned_reviewer_id = assigned_reviewer_id
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = change_cmd
+        self.score_category = score_category
+
+    def validate(self):
+        """Validates a suggestion object of type
```

Comment:
Perhaps, just say validates the SuggestionEditStateContent object.

Reply:
Done